src/smc-hub/client.coffee

```diff
@@ -1162,20 +1162,123 @@ class exports.Client extends EventEmitter
 
     mesg_invite_collaborator: (mesg) =>
         @touch()
+        #dbg = @dbg('mesg_invite_collaborator')
+        #dbg("mesg: #{misc.to_json(mesg)}")
         @get_project mesg, 'write', (err, project) =>
             if err
                 return
+            locals =
+                email_address : undefined
+                done          : false
+
             # SECURITY NOTE: mesg.project_id is valid and the client has write access, since otherwise,
             # the @get_project function above wouldn't have returned without err...
-            @database.add_user_to_project
-                project_id : mesg.project_id
-                account_id : mesg.account_id
-                group      : 'collaborator'  # in future will be "invite_collaborator", once implemented
-                cb         : (err) =>
-                    if err
-                        @error_to_client(id:mesg.id, error:err)
+            async.series([
+                (cb) =>
+                    @database.add_user_to_project
+                        project_id   : mesg.project_id
+                        account_id   : mesg.account_id
```

Comment:
I'm suddenly really, realliy worried about what will happen with Minerva (say) when they use the API to dynamically add hundreds of students to 20 projects for a session.    It seems like whether or not to send an email should be an option in the invite mesg.  The option is off by default for the API (or on, but we tell minerva to turn it off), but on for the browser client.  What do you think?

Reply:
yes, I didn't think about that. It should be off by default for the API.